src/main/scala/ch/epfl/bluebrain/nexus/kg/routes/CommonRoutes.scala

```diff
@@ -132,21 +144,79 @@ private[routes] abstract class CommonRoutes(
+  }
+
+  private def getResource(id: AbsoluteIri, schemaOpt: Option[Ref])(implicit output: NonBinaryOutputFormat): Route =
+    trace(s"get$resourceName") {
+      val idRes = Id(project.ref, id)
+      concat(
+        (parameter('rev.as[Long]) & noParameter('tag)) { rev =>
+          completeWithFormat(resources.fetch(idRes, rev, schemaOpt).materializeRun(id.ref, Some(rev), None))
+        },
+        (parameter('tag) & noParameter('rev)) { tag =>
+          completeWithFormat(resources.fetch(idRes, tag, schemaOpt).materializeRun(id.ref, None, Some(tag)))
+        },
+        (noParameter('tag) & noParameter('rev)) {
+          completeWithFormat(resources.fetch(idRes, schemaOpt).materializeRun(id.ref, None, None))
+        }
+      )
+    }
+
+  private def completeWithFormat(fetched: Future[Either[Rejection, (StatusCode, ResourceV)]])(
+      implicit output: NonBinaryOutputFormat): Route =
+    output match {
+      case f: JsonLDOutputFormat =>
+        implicit val format = f
+        complete(fetched)
+      case Triples =>
+        implicit val marshaller = stringMarshaller(Triples)
+        complete(fetched.map(_.map {
+          case (status, resource) =>
+            status -> resource.value.graph.triples
+              .map { case (s, p, o) => s"${s.show} ${p.show} ${o.show} ." }
+              .mkString("\n")
+        }))
+      case DOT =>
+        implicit val marshaller = stringMarshaller(DOT)
+        complete(fetched.map(_.map {
+          case (status, resource) =>
+            status -> resource.value.graph.asDot
+        }))
+    }
+
+  private def getFile(id: AbsoluteIri): Route =
+    trace("getFile") {
+      concat(
+        (parameter('rev.as[Long]) & noParameter('tag)) { rev =>
+          completeFile(resources.fetchFile(Id(project.ref, id), rev).value.runNotFound(id.ref))
+        },
+        (parameter('tag) & noParameter('rev)) { tag =>
+          completeFile(resources.fetchFile(Id(project.ref, id), tag).value.runNotFound(id.ref))
+        },
+        (noParameter('tag) & noParameter('rev)) {
+          completeFile(resources.fetchFile(Id(project.ref, id)).value.runNotFound(id.ref))
+        }
+      )
+    }
+
+  private def completeFile(f: Future[(FileAttributes, FileStore.AkkaOut)]): Route =
+    onSuccess(f) {
+      case (info, source) =>
+        val filename = urlEncodeOrElse(info.filename)("file")
+        (respondWithHeaders(RawHeader("Content-Disposition", s"attachment; filename*=UTF-8''$filename")) & encodeResponse) {
+          headerValueByType[Accept](()) { accept =>
+            val contentType = ContentType.parse(info.mediaType).getOrElse(Binary.contentType)
+            if (accept.mediaRanges.exists(_.matches(contentType.mediaType)))
+              complete(HttpEntity(contentType, info.bytes, source))
+            else
+              failWith(UnacceptedResponseContentType(
+                s"Binary Content-Type is '$contentType' and Accept HTTP Headers are '${accept.mediaRanges.mkString(", ")}'"))
```

Comment:
Replace `Binary Content-Type is '$contentType' and Accept HTTP Headers are '${accept.mediaRanges.mkString(", ")}'` with
`File media type '$contentType' does not match the Accept header value '${accept.mediaRanges.mkString(", ")}'.`